qsc/to_desc.py

```diff
@@ -0,0 +1,244 @@
+    )
+    # convert to DESC Fourier coefficients
+    m_pol, n_tor, R_mn = ptolemy_identity(RBS, RBC)
+    m_pol, n_tor, Z_mn = ptolemy_identity(ZBS, ZBC)
+
+    # write to DESC file
+    file_object = open(filename, "w+")
+    file_object.write(
+        "# This DESC input file was generated by pyQSC: github.com/landreman/pyQSC\n"
+    )
+    file_object.write(
+        "# Date: {}, Time: {}, UTC {}\n".format(
+            datetime.now().strftime("%B %d, %Y"),
+            datetime.now().strftime("%H:%M:%S"),
+            datetime.now().astimezone().strftime("%z"),
+        )
+    )
+    file_object.write(
+        "# Near-axis parameters:  radius r = {}, etabar = {}\n".format(r, self.etabar)
+    )
+    file_object.write(
+        "# nphi = {}, order = {}, sigma0 = {}, I2 = {}, B0 = {}\n".format(
+            self.nphi, self.order, self.sigma0, self.I2, self.B0
+        )
+    )
+    file_object.write(
+        "# Resolution parameters: ntheta = {}, ntorMax = {}\n".format(ntheta, ntorMax)
+    )
+    file_object.write("\n# global parameters\n")
+    file_object.write("sym = {}\n".format(sym))
+    file_object.write("NFP = {}\n".format(NFP))
+    file_object.write("Psi = {}\n".format(Psi))
+    file_object.write("\n# spectral resolution\n")
+    if "L_rad" in params.keys():
+        file_object.write(
+            "L_rad = " + " ".join(map(str, np.atleast_1d(params["L_rad"]))) + "\n"
+        )
+    file_object.write("M_pol = " + " ".join(map(str, np.atleast_1d(M_pol))) + "\n")
+    file_object.write("N_tor = " + " ".join(map(str, np.atleast_1d(N_tor))) + "\n")
+    if "L_grid" in params.keys():
+        file_object.write(
+            "L_grid = " + " ".join(map(str, np.atleast_1d(params["L_grid"]))) + "\n"
+        )
+    if "M_grid" in params.keys():
+        file_object.write(
+            "M_grid = " + " ".join(map(str, np.atleast_1d(params["M_grid"]))) + "\n"
+        )
+    if "N_grid" in params.keys():
+        file_object.write(
+            "N_grid = " + " ".join(map(str, np.atleast_1d(params["N_grid"]))) + "\n"
+        )
+    file_object.write("\n# continuation parameters\n")
+    if "bdry_ratio" in params.keys():
+        file_object.write(
+            "bdry_ratio = "
+            + " ".join(map(str, np.atleast_1d(params["bdry_ratio"])))
+            + "\n"
+        )
+    if "pres_ratio" in params.keys():
+        file_object.write(
+            "pres_ratio = "
+            + " ".join(map(str, np.atleast_1d(params["pres_ratio"])))
+            + "\n"
+        )
+    if "pert_order" in params.keys():
+        file_object.write(
+            "pert_order = "
+            + " ".join(map(str, np.atleast_1d(params["pert_order"])))
+            + "\n"
+        )
+    file_object.write("\n# solver tolerances\n")
+    if "ftol" in params.keys():
+        (
+            file_object.write(
+                "ftol = " + " ".join(map(str, np.atleast_1d(params["ftol"])))
+            )
+            + "\n"
```

Comment:
I think the order of these 2 lines is reversed? Same for the xtol, gtol, nfev bits in the next few lines.